src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployContext.kt

```diff
@@ -0,0 +1,87 @@
+package no.skatteetaten.aurora.boober.service
+
+import com.fasterxml.jackson.databind.node.ArrayNode
+import no.skatteetaten.aurora.boober.service.openshift.OpenShiftResponse
+import no.skatteetaten.aurora.boober.utils.openshiftKind
+import no.skatteetaten.aurora.boober.utils.openshiftName
+
+open class RedeployContext(private val openShiftResponses: List<OpenShiftResponse>) {
```

Comment:
This class has some duplication and some places the methods perform operations non lazily that could be lazy. say in getImageStream() if you wrap that block in a lazy it will only do it once. Since the data comming in is immutable this is ok.

Reply:
Could it be an option to simply initialize these variables when the instance is created? Just add them as class variables that are initilized with the values from openShiftResponses. Should work since the openShiftResponses are added in the constructor.

```
private val imageStream: OpenShiftResponse? = openShiftResponses.find { it.responseBody?.openshiftKind == "imagestream" }
    private val deploymentConfig: OpenShiftResponse? = openShiftResponses.find { it.responseBody?.openshiftKind == "deploymentconfig" }
```

Reply:
Pushed a suggestion for this in the latest commit